course5/homework6/string.cc

```diff
@@ -0,0 +1,106 @@
+
+#include "string.h"
+
+namespace String {
+
+String::String(const String& copy):
+	mContents(std::make_unique<char[]>(mSize)),
+	mSize(copy.mSize)
+{
+	for (unsigned idx = 0; idx < mSize; idx++)
+		mContents[idx] = copy.mContents[idx];
+}
+
+String::String(String&& move):
+	mContents(std::move(move.mContents)),
+	mSize(move.mSize)
+{
+}
+
+String& String::operator=(const String& copy)
+{
+	mSize = copy.mSize;
+	mContents = std::make_unique<char[]>(mSize);
+	for (unsigned idx = 0; idx < mSize; idx++)
+		mContents[idx] = copy.mContents[idx];
+	return *this;
+}
+
+String& String::operator=(String&& move)
+{
+	mSize = move.mSize;
+	mContents = std::move(move.mContents);
+	return *this;
+}
+
+String::String(const char* charString)
+{
+	*this = charString;
+}
+
+String& String::operator=(const char* lhs)
+{
+	// Find the end of the string
+	while (lhs[mSize] != '\0')
```

Comment:
I hadn't realised we were allowed to use cstring methods, this makes it indeed a lot easier!